semgrep-core/src/analyzing/Dataflow_svalue.ml

```diff
@@ -305,6 +305,7 @@ let rec is_symbolic_expr expr =
   match expr.G.e with
   | G.L _ -> true
   | G.N _ -> true
+  | G.IdSpecial _ -> true
```

Comment:
Why IdSpecial was added there? I'm just curious which example we were not handling before.
Do we need all IdSpecial or only a subset?

Reply:
Should have made this more clear, but the example I added from #3207 is one case, since we have `boolean b1 = !name.contains("../")` and the `!` is an `IdSpecial(Op)`. I think in general we want all the `IdSpecial` so that e.g. we can propagate `x = this.y`.